### ISSUE TITLE
[Beta 20151213] + Sign interferes with select

### ISSUE THREAD

**Issue by enoch85** [owncloud/android#1346]
When selecting files with the Android app you can't select the last ones because it interferes with the + sign to add. 

A possible soulution would be to inactivate the + sign when selecting.

**Phone**
OnePlus 2
Android 5.1.1

**Comment by przybylski** [comment]
Checkboxes should be gone long ago, and they should be replaced by long press selection

**Comment by enoch85** [comment]
@przybylski So you think it's better to delete every image once at the time, rather than have the possibility to select several at once? Sorry, don't agree with you on that one.

And if I recall it correctly, that's the behavior of the stable app.

**Comment by tobiasKaminsky** [comment]
@przybylski This is a new feature from branch multiSelect.

@enoch85 Your solution seems very good for me.
@AndyScherzinger how do we want to proceed with this? Can you create a new PR/branch which depends on "your" material_fab and "my" multiSelect, and implements the idea of @enoch85?

**Comment by przybylski** [comment]
@enoch85 what i meant was, we should use contextual action bar with list view like described here: https://androidperspective.wordpress.com/2013/04/17/contextual-action-bar-with-listview/ instead of putting checkboxes

**Comment by enoch85** [comment]
@przybylski Oh, that looks awesome. :) @AndyScherzinger What do you think, would it be possible?

**Comment by AndyScherzinger** [comment]
What would solve the problem and imho be the right thing to do anyways would be to ad ennough whitespace at the end of a list so that the list items can scroll completely above the FAB when the user scrolls to the end of the list. At least that is what I am used to in apps utilizing the FAB, e.g. SeriesGuide. 

So I would change that for starters.
Besides this small fix, for platform expected behavior we should move to the  behavior referenced by @przybylski or even further sine the described behavior would be >= ICS while material design lists basically make the icon on the left (our folder/file icons) an animated tickbox (e.g. like it is done in gmail). But that is a whole other load of work. So I think we should split it into these two parts. :)

**Comment by jancborchardt** [comment]
I’ve been in favor of moving to »long-press for select« since a long time, see https://github.com/owncloud/android/issues/899 ;)
(@enoch85 for info: Long-press will initiate the multiselect mode, every file after that only has to be tapped to be selected, not long-pressed.)

Just needs someone to implement and review it. As @tobiasKaminsky mentioned there is a branch, see the PR at https://github.com/owncloud/android/pull/1176

**Comment by jancborchardt** [comment]
So @davivel @masensio since this causes a bug where you can not select the last file, we should push forward https://github.com/owncloud/android/pull/1176

**Comment by tobiasKaminsky** [comment]
Of course you can select the last item:
If there is no entry selected so far --> long press
if there is already multi-select --> just click on any point of the item (can also be very left)

The checkbox is just there to visualise the selection state, but it is not necessary to tip exactly on it.

**Comment by enoch85** [comment]
@tobiasKaminsky As it is possible to select, it's not obvious.

**Comment by AndyScherzinger** [comment]
Like I said this should be done via being able scroll beyond the FAB and that would simple fix this matter and would be the right thing to do anyways ;)

**Comment by AndyScherzinger** [comment]
I also pushed the change to the beta branch, see attached the screenshots from the actual state on the fab branch:
![device-2015-12-15-172001](https://cloud.githubusercontent.com/assets/1315170/11816434/be7b5b08-a350-11e5-8d98-e1512cf50a97.png)

![device-2015-12-15-172125](https://cloud.githubusercontent.com/assets/1315170/11816438/c244cd50-a350-11e5-8cf5-d2b4edede76d.png)

Feel free to comment. The general show/hide would need to be discussed and detailed. I also thought about it and it could also be some show/hide when scrolling down or up and things like that so I would leave that for now and go with this solution.

**Comment by enoch85** [comment]
I think it's too much whitespace at the botton. Why not just remove it while selecting? When you are selecting you aren't selecting because you want to upload something, you select because you want to edit/delete/move the files you've selected, IMHO.

**Comment by AndyScherzinger** [comment]
The whitespace is material standard spacing (spacing to the right and bottom - and now also to the top (end of the list). 

The button needs to be below the whole thing at some point anyways. As soon as someone finds the time to actually start implementing the three-dot menu per list item (positioned in the bottom right corner) we would have to scroll past the list anyways b/c else you would end up with your described problem but for the list item menu click. So hiding the fab won't work for that scenario.

**Comment by enoch85** [comment]
@AndyScherzinger A bit tired now, hard to imagine, but I'm sure it will be nice. :)

**Comment by AndyScherzinger** [comment]
@enoch85 This might give you a rough idea:
https://lh3.googleusercontent.com/Qw9I4Vfxh30PNXJVGRqtybamXPdg50xmIiSMB7qbXqflzxHHDL92fLB8RguM1QHEXg=h900

This is how SeriesGuide handles it, even though that SeriesGuide is using tiles while we might stay with standard list items (design wise) but it will kind of have a similar layout.

**Comment by enoch85** [comment]
@AndyScherzinger Ok, so that means that 2/5 is useless? It looks like you only can access 3, the rest is blocked by a big fab(?)

**Comment by AndyScherzinger** [comment]
Na... When you scroll down the list has enough whitespace at the end so the last can be scrolled above the fab, like in my screen shots

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger I thought that @jancborchardt and me have decided to not implement the "per item three dotted menu" and rather use the way I have implemented it now with the multi-select.
Otherwise the #1176 PR is useless ;)

**Comment by AndyScherzinger** [comment]
Hmm, That seemed to slipped through for me. Went back to the tickets and you are right, this is all covered by your PR. But that would then also mean the discussion about the new changed menu layout doesn't make any sense then. @jancborchardt?

**Comment by jancborchardt** [comment]
@tobiasKaminsky @AndyScherzinger I’m actually not sure where we stood with the 3-dots-menu for each item. I still think it makes sense in the list view. In grid view we indeed need another way (maybe the controls only being displayed on scroll in the bottom right and fading out again). But needing long press to access the actions for a single item is very non-material.

**Comment by jancborchardt** [comment]
And the multiselect PR #1176 is to fix the current behavior, eh?

**Comment by tobiasKaminsky** [comment]
Multiselect can solve everything ;)
I do not know in which situation it is useful to have the "three dots" menu in each line?
If I want to edit one item, I select it with a long click and use the "three dots" menu in the top.
Same is true for selecting multiple items.
(and it is the way the standard gallery app handles it)

**Comment by AndyScherzinger** [comment]
I think @jancborchardt's point here is efficiency :)
* multiselect implementation: long click an item, click action bar three dot, choose action
* three dot per item: click three dot, choose action

--> 2 clicks Vs 3 clicks (while the first is also a long click)

To do it material style we maybe should mark items click on the left aligned item icon like gmail and other do it. Ling click at least to me seems to be rather legacy :)

**Comment by tobiasKaminsky** [comment]
At least for the grid view we cannot copy the behaviour of gmail but have a look at gallery.
And I do not think it is a good idea to have two different behaviours for list or grid view.

Long click: legacy --> again: galley and other apps are doing it.

Your idea about "clicking on the left item" is also only possible in list view and not in grid view...

**Comment by AndyScherzinger** [comment]
True, I forgot about the Grid View.... So yeah, the long press multi select also seems to me the best as in most consistent solution. :+1: Sorry for the confusion!

**Comment by tobiasKaminsky** [comment]
No problem. I think it is important to discuss the pro and contras of a solution and therefore everything is fine :+1:

**Comment by tobiasKaminsky** [comment]
So...
@AndyScherzinger has solved this issue by adding a padding on the bottom?
Can I close this? @enoch85

**Comment by enoch85** [comment]
@tobiasKaminsky Yeah, as long as you solve it I'm fine with closing. :) :+1:

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky that's how I fixed it, yes! :)

**Comment by tobiasKaminsky** [comment]
Yeah, all credits belong to @AndyScherzinger :+1:

**Comment by jancborchardt** [comment]
The point about the 3-dots-icon is less about number of clicks than about discoverability. When the icon is not there, it's very difficult to figure out how to e.g. share a file. Long-click is usually only used for multiselect, not to initiate a basic action.

I'd say it while the long-press approach should work, and is the way to do it for the grid view (at least for now), we can improve the UX in the list view by showing the share and 3-dots-menu directly. Exactly as in the mobile view of the web interface.

**Comment by tobiasKaminsky** [comment]
@jancborchardt This is fine for me, but is also more code to maintain and handle it...

**Comment by jancborchardt** [comment]
@tobiasKaminsky I know that, but however since this is very basic functionality, I would say this is quite important.

**Comment by tobiasKaminsky** [comment]
I have created a new issue regarding this: #1396
Therefore I close this one.
